### PR TITLE
Replaced dependency on DotNetZip with System.IO.Compression.ZipFile

### PR DESCRIPTION
**Pull request by tombogle**
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/libpalaso/1360)

### CODE REVIEW

**Review by hatton — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/libpalaso/1360)* status: 0 of 11 files reviewed, 1 unresolved discussion (waiting on @andrew-polk, @josephmyers, and @tombogle)

___
*[a discussion](https://reviewable.io/reviews/sillsdev/libpalaso/1360#-OB85JX4CrfA23n2PCsj:-OB85JX4CrfA23n2PCsk:b-oh3mak) (no related file):*
What problem is this change solving?
Can you confirm that the unit tests covered corner cases like non-ascii characters?

**Comment by tombogle** [comment]
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

What problem is this change solving?
Can you confirm that the unit tests covered corner cases like non-ascii characters?
</blockquote></details>

The DotNetZip package is now deprecated. Not sure that it actually represented any vulnerability or problem, but since we're making a bunch of other breaking changes and updating dependencies, this seemed like the right time to bite the bullet and switch to a package that is going to be actively maintained. I don't think any of the existing unit tests checked for handling special characters, but I just tweaked one of them to prove that it works. Not 100% sure that's what you had in mind.

**Review by josephmyers — CHANGES_REQUESTED** [review]
The code overall looks good, thanks for doing this. But take a look at the test failures. They look legitimate. Some version conflicts, probably due to the changed references

File: SIL.Archiving/RampArchivingDlgViewModel.cs [hunk 1]
```diff
@@ -1674,41 +1674,34 @@ private void CreateZipFile(CancellationToken cancellationToken)
 
 			ReportMajorProgressPoint(StringId.SavingFilesInPackage, cancellationToken);
 
-			using (var zip = new ZipFile())
+			using (var zip = ZipFile.Open(PackagePath, ZipArchiveMode.Create))
 			{
-				// RAMP packages must not be compressed or RAMP can't read them.
```
Comment (hahn-kev) [inline SIL.Archiving/RampArchivingDlgViewModel.cs:1679]: have we tested the zip files made with this change to ensure RAMP can read them? it sounds like it's picky

**Review by hahn-kev — APPROVED** [review]
looks good to me, I left a comment about RAMP as there was an old comment which was lost about it being picky about the zip file

File: SIL.Archiving/RampArchivingDlgViewModel.cs (hunk 1 above)
Comment (tombogle) [reply SIL.Archiving/RampArchivingDlgViewModel.cs:1679]: I just tested it and RAMP seems to be completely happy with the package as zipped.

**Comment by tombogle** [comment]
@hatton @hahn-kev, if you guys are happy, please go ahead and release this PR from purgatory.

**Review by hatton — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/libpalaso/1360)* status: 0 of 11 files reviewed, 2 unresolved discussions (waiting on @andrew-polk, @hahn-kev, and @tombogle)

___
*[a discussion](https://reviewable.io/reviews/sillsdev/libpalaso/1360#-OB85JX4CrfA23n2PCsj:-OC0SVhk4-h4K28x14gK:bvmc19z) (no related file):*
<details><summary><i>Previously, tombogle (Tom Bogle) wrote…</i></summary><blockquote>

The DotNetZip package is now deprecated. Not sure that it actually represented any vulnerability or problem, but since we're making a bunch of other breaking changes and updating dependencies, this seemed like the right time to bite the bullet and switch to a package that is going to be actively maintained. I don't think any of the existing unit tests checked for handling special characters, but I just tweaked one of them to prove that it works. Not 100% sure that's what you had in mind.
</blockquote></details>

Sorry, I don't have the margin to follow up and look at details here. I just wanted to raise the issue.

**Review by hahn-kev — APPROVED** [review]
